src/con_duct/__main__.py

```diff
@@ -149,16 +151,24 @@ def aggregate(self, other: ProcessStats) -> ProcessStats:
                 lgr.debug(f"using {other.cmd}.")
                 cmd = other.cmd
             lgr.debug(f"using {self.cmd}.")
+
+        self.stat.update(other.stat)
         return ProcessStats(
             pcpu=max(self.pcpu, other.pcpu),
             pmem=max(self.pmem, other.pmem),
             rss=max(self.rss, other.rss),
             vsz=max(self.vsz, other.vsz),
             timestamp=max(self.timestamp, other.timestamp),
             etime=other.etime,  # For the aggregate always take the latest
+            stat=self.stat,
             cmd=cmd,
         )
 
+    def for_json(self) -> dict:
```

Comment:
could it be used for yaml? or in general internally as a dict model?  my point is, why it is not `as_dict`?  or may be there is just a matter or adding "wrapper" to Counter so it automagically presents itself as a dict when `dict` on ProcessStats is called?

Reply:
For this I chose `for_json`, simply to match `Sample.for_json`.

I don't feel strongly from the name, but I think having `as_dict` and `asdict` would be confusing. The nice thing about `for_json` is that it is explicit that the the return value will be serializable all the way down. 

I took a look at how `asdict` works under the hood, and `_as_dict_inner` is called recursively, but only for other instances of dataclasses. 
https://github.com/python/cpython/blob/3.12/Lib/dataclasses.py#L1335-L1347

In our case, we are just getting a deepcopy of the Counter value, so theres no "hook" to override.
 https://github.com/python/cpython/blob/3.12/Lib/dataclasses.py#L1386

I suppose we could make a CounterDataclass, but that feels like overkill when `for_json` is so simple

Reply:
why `as_dict` or `asdict` would be confusing?  isn't it what it does?

but fine with me in general. I just find `for_json` confusing since there is little nothing specific to json there.